src/javaharness/java/arcs/api/PECInnerPortImpl.java

```diff
@@ -1,27 +1,32 @@
 package arcs.api;
 
-import javax.inject.Inject;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.function.Function;
 
 public class PECInnerPortImpl implements PECInnerPort {
+    private String id;
     private ShellApi shellApi;
     private ParticleExecutionContext pec;
     private ThingMapper mapper;
     private StorageProxyFactory storageProxyFactory;
+    private PortableJsonParser jsonParser;
 
-    @Inject
-    public PECInnerPortImpl(ShellApi shellApi, ParticleExecutionContext pec) {
-      this.shellApi = shellApi;
-      this.pec = pec;
-      this.mapper = new ThingMapper("j");
+    public PECInnerPortImpl(String id,
+                            ShellApi shellApi,
+                            ParticleExecutionContext pec,
+                            PortableJsonParser jsonParser) {
+        this.id = id;
+        this.shellApi = shellApi;
+        this.pec = pec;
+        this.mapper = new ThingMapper("j");
+        this.jsonParser = jsonParser;
     }
 
     @Override
     public void handleMessage(PortableJson message) {
         String messageType = message.getString("messageType");
         PortableJson messageBody = message.getObject("messageBody");
-        String identifier = messageBody.getString("identifier");
```

Comment:
Why did you move and inline this?

Reply:
because SimpleCallback (and other future calls to come) don't have this field in the message.